packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -548,6 +555,18 @@ class ProfileMacOSBundleFlutterAssets extends MacOSBundleFlutterAssets {
     CompileMacOSFramework(),
     ProfileUnpackMacOS(),
   ];
+
+  @override
+  List<Source> get inputs => <Source>[
+    ...super.inputs,
+    const Source.pattern('{BUILD_DIR}/App.framework.dSYM/Contents/Resources/DWARF/App'),
```

Comment:
`CompileMacOSFramework` already builds this and defines it as an output on line 339 
https://github.com/vaind/flutter/blob/b6ceb546f64a9b492a362e07ba643fe3a942567b/packages/flutter_tools/lib/src/build_system/targets/macos.dart#L339

This task here (and the one in the comment below) collect it as an input and moves to the `OUTPUT_DIR`, same as the rest of the (base class) `MacOSBundleFlutterAssets`: 
https://github.com/flutter/flutter/blob/b6ceb546f64a9b492a362e07ba643fe3a942567b/packages/flutter_tools/lib/src/build_system/targets/macos.dart#L352-L362

Reply:
Oh, I see, you're working around the issue that we don't have a way to specify a build mode version of `Source`, otherwise it would be in the `MacOSBundleFlutterAssets` baseclass, but just for profile and release.